django/library/serializers.py

```diff
@@ -110,19 +190,46 @@ def get_profile_url(self, instance):
             )
 
     def update(self, instance, validated_data):
-        affiliations = validated_data.pop("affiliations", None)
-        validated_data.pop("given_name", None)
-        validated_data.pop("family_name", None)
+        if not self._is_not_used_or_used_by_current_release_only(instance):
+            if self._trying_to_modify_attributes(instance, validated_data):
+                raise ValidationError(
+                    _(
+                        "Contributors can only be updated when they are exclusive to one codebase"
+                    )
+                )
+        # Server side validation:
+        # if a contributor is made from a user, do not allow to modify it's data.
+        # POP following:
+        # 'given_name'
+        # 'middle_name'
+        # 'family_name'
+        # 'email'
+        # 'type'
+        # 'affiliations'
+        # 'json_affiliations'
+        user_id  = validated_data.pop("user_id", None)
```

Comment:
Since this is from the `update()` action and this only deals with the case of updating a Contributor which was created from an existing user, I think we shouldn't allow changing Contributor-AuthUser relation (user_id) in this case, and also any of the other user related attributes for the sake of consistency of Contributor-AuthUser fields. .
When a new Contributor is created from an existing AuthUser the user_id will be set inside the `create()` action which gets the whole `user` as parameter.

Reply:
Does it remove the existing user relation from the contributor? It reads like that but idk if that actually happens

Reply:
no, it doesn't remove the user_id from contributor